### DISCUSSION TITLE
Is it possible to create a k8s resource directly?

### DISCUSSION THREAD

**Discussion by Mau04** [argoproj/argo-workflows#4841] [Q&A] [answered]
Hi all!
We are currently evaluating Argo Workflows for our use case. We'd like to create a kubernetes resource as part of a workflow as described [here](https://argoproj.github.io/argo/examples/#kubernetes-resources). We noticed that this is done in an own pod. Is it possible to configure it in a way, that the additional pod is not spawned and the resource is deployed to the cluster directly or is there a specific reason for doing that inside a pod?

**Comment by VictoryAnn** [comment]
Hi, 
We have the same question. 
If we try to create a k8s resource directly, how to update the status of workflow. I notice that argoexec pod watch status of k8s resource.  
Is there a way to create a k8s resource and update workflow status without argoexec pod.

**Comment by alexec** [comment, marked as answer]
It is not possible to create a Kubernetes resource directly. This is by design. The workflow pod creates the resource using the service account specified in the workflow, which needs the `create your-resource` role to do this. If the controller were to do this, it would need that role in every namespace (cluster scoped installs), this is a lot more power for the controller that it actually needs.

**Reply by agilgur5** [reply]
See also a more detailed answer in #4712, which this discussion duplicates.
